websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,166 @@
+const presence = new Presence({
+		clientId: "580032576434077707",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video = {
+	current: 0,
+	duration: 0,
+	paused: true,
+};
+
+presence.on(
+	"iFrameData",
+	(data: { current: number; duration: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		pth = document.location.pathname;
+	if (document.location.host === "desu-online.pl") {
+		presenceData.smallImageKey =
+			"https://cdn-icons-png.flaticon.com/512/5611/5611971.png";
+		if (document.location.search !== "") {
```

Comment:
You use a lot of properties on `document.location`.
Consider refactoring by using destructuring.
Something like this:
```ts
const { pathname, host, search } = window.location;
```